diff --git a/src/Elastic/Indexer/Rebuilder/Rebuilder.php b/src/Elastic/Indexer/Rebuilder/Rebuilder.php
index d8280557a..6aa79134b 100644
--- a/src/Elastic/Indexer/Rebuilder/Rebuilder.php
+++ b/src/Elastic/Indexer/Rebuilder/Rebuilder.php
@@ -374,8 +374,9 @@ class Rebuilder {
 		// Cannot be altered by a simple settings update and requires a complete
 		// rebuild
 		unset( $indexDef['settings']['number_of_shards'] );
+ 		unset( $indexDef['settings']['index.knn'] );
 
		// #4341
 		// ES 5.6 may cause a "Can't update [index.number_of_replicas] on closed
 		// indices" see elastic/elasticsearch#22993 and should be fixed with ES 6.4.
 		if ( !$this->client->isOpenSearch() && version_compare( $this->client->getVersion(), '6.4.0', '<' ) ) {
